### PR TITLE
Update virtual environment guide URL on contributing page

### LINKED ISSUE #2199: Fix broken virtual environment guide link on contributing page
**Describe the bug**
On the contributing page, the link to the astropy page on virtual environments does not work.

**To Reproduce**
1. Go to the [Basics section](https://pvlib-python.readthedocs.io/en/stable/contributing.html#the-basics)
2. Click on [astropy guide](http://astropy.readthedocs.org/en/latest/development/workflow/virtual_pythons.html)

**Expected behavior**
I think the correct URL is https://astropy-astrofrog.readthedocs.io/en/latest/development/workflow/virtual_pythons.html but could someone confirm?
[This link](https://pvlib-python.readthedocs.io/en/stable/user_guide/installation.html#set-up-a-virtual-environment) to the pvlib user guide section on setting up a virtual environment could also be integrated into the same sentence.

**Versions:**
Checked 10.5, 11.0 (stable), and 11.1 (latest) --- link not working

### PR DESCRIPTION
**Pull request by RDaxini**
- [X] Closes #2199
 - [X] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [ ] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [X] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [X] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

Original link: https://docs.astropy.org/en/latest/development/workflow/virtual_pythons.html
New link: https://astropy-astrofrog.readthedocs.io/en/latest/development/workflow/virtual_pythons.html
Could someone confirm the new link here is the correct/intended webpage for the guide?

### CODE REVIEW

**Comment by RDaxini** [comment]
I am wondering whether it might be also be useful to integrate [this link](https://pvlib-python.readthedocs.io/en/stable/user_guide/installation.html#set-up-a-virtual-environment), which directs to the pvlib user guide section on virtual environments, into the same sentence?

**Comment by kandersolar** [comment]
>  New link: https://astropy-astrofrog.readthedocs.io/en/latest/development/workflow/virtual_pythons.html
> Could someone confirm the new link here is the correct/intended webpage for the guide?

Based on the URL, this seems to be an "unofficial" set of docs that `astrofrog` (an astropy developer) has set up.  I don't think we should link to it.  

How about something like https://learn.scientific-python.org/development/tutorials/dev-environment/ instead?  I haven't compared in depth with the previous page, but in general I'd prefer to link to some kind of central resource like `scientific-python.org` instead of another package's docs.

**Comment by mikofski** [comment]
No, I would keep the sections separate. Although they are both related to setting up a dev environment for contributing, the 2nd focuses on using conda . I think pvlib needs to have its own help on this that doesn’t depend on a link to another project. There should be links between the two sections tho.

**Comment by RDaxini** [comment]
> No, I would keep the sections separate. Although they are both related to setting up a dev environment for contributing, the 2nd focuses on using conda . I think pvlib needs to have its own help on this that doesn’t depend on a link to another project. There should be links between the two sections tho.

Ah... when I said integrate the link into the sentence, I meant literally only the link --- not bring over the linked section and merge both sections.
See [b298674](https://github.com/pvlib/pvlib-python/pull/2200/commits/b298674dc6da2ca7869141d3d0b707b331d10aa7) for what I had in mind. Let me know what you think

**Review by kandersolar — APPROVED** [review]
LGTM!  I'll leave this open for a few days in case any other ideas pop up.

**Review by AdamRJensen — APPROVED** [review]

File: docs/sphinx/source/contributing.rst [hunk 1]
```diff
@@ -69,11 +69,13 @@ Requests <https://help.github.com/articles/using-pull-requests/>`_.
 
 We strongly recommend using virtual environments for development.
 Virtual environments make it trivial to switch between different
```
Comment (AdamRJensen) [inline docs/sphinx/source/contributing.rst:71]:
```suggestion
Virtual environments make it easy to switch between different
```
It honestly isn't trivial for new users

Comment (RDaxini) [reply docs/sphinx/source/contributing.rst:71]: True (wasn't my wording btw)
how about "easier" rather than "easy" though?

**Comment by RDaxini** [comment]
> LGTM! I'll leave this open for a few days in case any other ideas pop up.

Actually something I have been thinking about is that there is no obvious point of contact on the contributing page in case someone wants to reach out. What is the best point of contact, is it the Google group? Is there an email address? The Google group is linked in the [Easy ways to contribute](https://pvlib-python.readthedocs.io/en/stable/contributing.html#easy-ways-to-contribute) section, which is good, but perhaps referring to that link (or whatever is the best contact option) somewhere more explicitly as a means of getting in touch with the community for advice/questions about contributing to pvlib specifically might be an improvement.

Another thought, after reading some of the contributing pages on other projects (through the discussion in #2202), I feel like our contributing page would benefit from being broken up into a few pages. These pages would still all be under the same "Contributing" section, but they could be navigated through using the sidebar (like the user guide?)
In this way, we could keep the really basic basics one one (the first) page, then move other details such as code style, testing, virtual environments, etc. onto separate pages rather than just separate sections on the same page. I think that this single page in its current form may be overwhelming for new users; it certainly was for me not so many months ago 😅

What do you guys think? These points may be out of scope for this URL update PR (at least the second one is for sure) and may warrant a separate issue for further discussion first. Should I open a new issue for either or both of these points?

**Comment by cwhanse** [comment]
> What is the best point of contact, is it the Google group? Is there an email address? 

I would prefer that they open an Issue or Discussion rather than send an email.

**Comment by cwhanse** [comment]
> contributing page would benefit from being broken up into a few pages. 

Agree fully.

**Comment by RDaxini** [comment]
> > LGTM! I'll leave this open for a few days in case any other ideas pop up.
> 
> Actually something I have been thinking about is that there is no obvious point of contact on the contributing page in case someone wants to reach out. What is the best point of contact, is it the Google group? Is there an email address? The Google group is linked in the [Easy ways to contribute](https://pvlib-python.readthedocs.io/en/stable/contributing.html#easy-ways-to-contribute) section, which is good, but perhaps referring to that link (or whatever is the best contact option) somewhere more explicitly as a means of getting in touch with the community for advice/questions about contributing to pvlib specifically might be an improvement.
> 
> Another thought, after reading some of the contributing pages on other projects (through the discussion in #2202), I feel like our contributing page would benefit from being broken up into a few pages. These pages would still all be under the same "Contributing" section, but they could be navigated through using the sidebar (like the user guide?) In this way, we could keep the really basic basics one one (the first) page, then move other details such as code style, testing, virtual environments, etc. onto separate pages rather than just separate sections on the same page. I think that this single page in its current form may be overwhelming for new users; it certainly was for me not so many months ago 😅
> 
> What do you guys think? These points may be out of scope for this URL update PR (at least the second one is for sure) and may warrant a separate issue for further discussion first. Should I open a new issue for either or both of these points?

These points will be resolved separately now (#2210) so, other than those points, if there's anything else required for this PR then just let me know

**Comment by kandersolar** [comment]
Thanks @RDaxini!

**Pull request merged**